File: (Java)OrderdBroadcastReceiver/app/src/main/java/com/dilip/javaorderedbroadcastreceiver/MainActivity.java
package com.dilip.javaorderedbroadcastreceiver;

import android.content.IntentFilter;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    MyDemoReceiver myDemoReceiver = new MyDemoReceiver();
    MyDemoReceiver2 mReceiver = new MyDemoReceiver2();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        IntentFilter intentFilter = new IntentFilter("com.dilip.javaorderedbroadcastreceiver.ACTION_SEND");
        intentFilter.setPriority(80);
        registerReceiver(mReceiver, intentFilter);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(mReceiver);
    }

    public void sendBroadcast(View view) {

    }
}

File: Custom Broadcast Receiver/HiReceiver/app/src/main/java/com/dilip/receiver/MainActivity.kt
package com.dilip.receiver

import android.content.IntentFilter
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    val customBroadcast = CustomBroadcast()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val filter = IntentFilter("com.example.receiver.EXAMPLE_ACTION")
        registerReceiver(customBroadcast, filter)
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(customBroadcast)
    }
}